pkg/api/types.go

```diff
@@ -1443,6 +1461,14 @@ type VolumeMount struct {
 	SubPath string
 }
 
+// VolumeDevice describes a mounting of a raw block device within a container.
+type VolumeDevice struct {
+	//Required: This must match the Name of a Volume in the pod.
+	Name *string
```

Comment:
These two fields should not be pointers since they're required.

Reply:
@msau42 - I thought all new fields should be pointers?  @liggitt isn't  that what you said or did I misunderstand?

Reply:
if you need to be able to distinguish between set and unset (for purposes of defaulting), or need to be able to completely omit a struct field from serialization, then it has to be a pointer. this does not